### PR TITLE
Updated c++ coding standards example

### PR DESCRIPTION
**Pull request by nicholasluimy**
Updated coding standards example.
Made use of custom parsing in markdown.js to enable emphasis of code in <pre lang> block

### CODE REVIEW

**Comment by nicholasluimy** [comment]
yikes misnamed, should be referring to #160

**Comment by wkurniawan07** [comment]
You know, you can edit the PR's title and description after making the PR :p The only thing you can't salvage is when you named the branch wrongly, fortunately you did that correctly.

File: contents/coding-standards-cpp.md [hunk 1]
```diff
@@ -594,10 +593,9 @@ e.g. `testLogic_addTask_nullParameters_errorMessageExpected()`
     void processItem(const Item& item) {
         try {
             if (/* some test failed */) {
-                throw Exception();
+                /*bold*/throw Exception()/*/bold*/;
```
Comment (wkurniawan07) [inline contents/coding-standards-cpp.md]: Now `throw Exception()` are the same in both examples. Unfortunately I don't have enough C++ knowledge to tell the "right" or "wrong" way to throw Exceptions, so I'll need another help here @damithc. Or @nicholasluimy maybe you know?

**Comment by wkurniawan07** [comment]
One more thing, can you stage your changes in `gh-pages` so I can see how it looks like?

**Comment by nicholasluimy** [comment]
@wkurniawan07 http://nicholasluimy.github.io/website/contents/coding-standards-cpp.html

**Comment by damithc** [comment]
Let me check about the code example with a C++ expert. The rule mentions both throwing and catching, but the code example doesn't seem to illustrate both.

File: contents/coding-standards-cpp.md [hunk 2]
```diff
@@ -594,10 +593,9 @@ e.g. `testLogic_addTask_nullParameters_errorMessageExpected()`
...
             }
-        }
-        catch (Exception excp) {
+        } /*bold*/catch (Exception excp)/*/bold*/ {
             // Process excp
             //
```
Comment (damithc) [inline contents/coding-standards-cpp.md]: remove this extra // on both sides

File: contents/coding-standards-cpp.md (hunk 1 above)
Comment (damithc) [inline contents/coding-standards-cpp.md]: Henry gave me this example.
```c
try {
   if (/* some test failed */) {
      throw /*bold*/new/*/bold*/ Exception();  //throw by pointer
   }
} catch (Exception *excp) {  //catch by pointer
   // Process excp
}
```

File: contents/coding-standards-cpp.md [hunk 3]
```diff
@@ -594,10 +593,9 @@ e.g. `testLogic_addTask_nullParameters_errorMessageExpected()`
     void processItem(const Item& item) {
         try {
             if (/* some test failed */) {
-                throw Exception();
+                /*bold*/throw Exception()/*/bold*/;
             }
-        }
-        catch (Exception excp) {
+        } /*bold*/catch (Exception excp)/*/bold*/ {
             // Process excp
```
Comment (damithc) [inline contents/coding-standards-cpp.md]: Change to `// Process exception`  on both sides

File: scripts/markdown.js [hunk 4]
```diff
@@ -18,6 +18,8 @@ $.ajax({
             success: function(data) {
                 data = data.replace(/<th align="center">Good<\/th>/g, '<th align="center" class="example-good">Good</th>');
                 data = data.replace(/<th align="center">Bad<\/th>/g, '<th align="center" class="example-bad">Bad</th>');
+                data = data.replace(/\/\*bold\*\//g, '<b>');
+                data = data.replace(/\/\*\/bold\*\//g, '</b>');
```
Comment (damithc) [inline scripts/markdown.js]: Instead of bold, use a css class to control the look? e.g. highlight the text

**Comment by wkurniawan07** [comment]
@nicholasluimy you can work on the comments added

**Comment by nicholasluimy** [comment]
@wkurniawan07 i've updated the page as based off the comments.

@damithc prof, i'm not sure how to use a css class to bold it instead, i think it's working funnily because of the pre tags wrapping that code segment to give it syntax highlighting, which is why i ended up using that hackish method to bold it using javascript

File: contents/coding-standards-cpp.md (hunk 1 above)
Comment (wkurniawan07) [inline contents/coding-standards-cpp.md]: @nicholasluimy do update the examples as illustrated above.

File: params.json [hunk 5]
```diff
@@ -0,0 +1 @@
...g your page, you can revisit the page generator and switch to another theme. Your Page content will be preserved.\r\n\r\n### Creating pages manually\r\nIf you prefer to not use the automatic generator, push a branch named `gh-pages` to your repository to create a page manually. In addition to supporting regular HTML content, GitHub Pages support Jekyll, a simple, blog aware static site generator. Jekyll makes it easy to create site-wide headers and footers without having to copy them across every page. It also offers intelligent blog support and other advanced templating features.\r\n\r\n### Authors and Contributors\r\nYou can @mention a GitHub username to generate a link to their profile. The resulting `<a>` element will link to the contributor’s GitHub Profile. For example: In 2007, Chris Wanstrath (@defunkt), PJ Hyett (@pjhyett), and Tom Preston-Werner (@mojombo) founded GitHub.\r\n\r\n### Support or Contact\r\nHaving trouble with Pages? Check out our [documentation](https://help.github.com/pages) or [contact support](https://github.com/contact) and we’ll help you sort it out.\r\n","google":"","note":"Don't delete this file! It's used internally to help with page regeneration."}
```
Comment (wkurniawan07) [inline params.json]: This shouldn't be part of the PR.

Comment (wkurniawan07) [inline params.json]: The same for the new `.css` files added.

**Comment by damithc** [comment]
Closing because the issue is no longer relevant. Thanks for your effort @nicholasluimy
